compute_tools/src/monitor.rs

```diff
@@ -106,6 +106,27 @@ fn watch_compute_activity(compute: &ComputeNode) {
                         continue;
                     }
                 }
+                //
+                // Do not suspend compute if autovacuum is running
+                //
+                let autovacuum_count_query = "select count(*) from pg_stat_activity where backend_type = 'autovacuum worker'";
```

Comment:
I am not sure, do we need to add backendDesc 'autovacuum launcher' to the predicate https://github.com/neondatabase/postgres/blob/9d38474fd52046fb77b4f5e3b121e346dbcd5c89/src/backend/utils/init/miscinit.c#L259C26-L259C26

`where backend_type in ('autovacuum worker', 'autovacuum launcher')`

We could also early out in the scan on the pg_stat_activity view using

`select 1 from pg_stat_activity where backend_type in ('autovacuum worker', 'autovacuum launcher') limit 1`

and then determine whether the query returns 0 or 1 rows. 
This would be faster because we don't need to aggregate and can exit the scan once the row is found.
This may not be relevant if you only have 10 connections running but what if you have a compute with 1000 open connections.

Reply:
> I am not sure, do we need to add backendDesc 'autovacuum launcher' to the predicate

Certainly not. Autovacuum launcher is always alive.

Reply:
I had checked that before asking: Not on my compute node:

```
select backend_type from pg_stat_activity;
neondb=> select backend_type from pg_stat_activity;
-[ RECORD 1 ]+---------------
backend_type | 
-[ RECORD 2 ]+---------------
backend_type | 
-[ RECORD 3 ]+---------------
backend_type | 
-[ RECORD 4 ]+---------------
backend_type | 
-[ RECORD 5 ]+---------------
backend_type | client backend
-[ RECORD 6 ]+---------------
backend_type | 
-[ RECORD 7 ]+---------------
backend_type | 
-[ RECORD 8 ]+---------------
backend_type | 
-[ RECORD 9 ]+---------------
backend_type | 
-[ RECORD 10 ]---------------
backend_type | 
-[ RECORD 11 ]---------------
backend_type | 
-[ RECORD 12 ]---------------
backend_type | 
-[ RECORD 13 ]---------------
backend_type | 
``

Reply:
Locating first record can save traversal of 1/2 records in average.
Taken in account that pg_stat_activity is actually virtual table, scaling this table even for large number of backend can not somehow affect performance.